api-parent/consumption-api/src/main/java/com/sap/cloud/environment/servicebinding/api/ServiceIdentifier.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+package com.sap.cloud.environment.servicebinding.api;
+
+import javax.annotation.Nonnull;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Represents an <b>equatable</b> service identifier that can be used across applications.
+ */
+public final class ServiceIdentifier
+{
+    @Nonnull
+    private static final Map<String, ServiceIdentifier> INSTANCES = new ConcurrentHashMap<>();
+
+    /**
+     * Represents the <a href="https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/overview">SAP Destination
+     * Service (Cloud Foundry)</a>.
+     */
+    @Nonnull
+    public static final ServiceIdentifier DESTINATION = of("destination");
+
+    /**
+     * Represents the <a href=
+     * "https://help.sap.com/docs/connectivity/sap-btp-connectivity-cf/what-is-sap-btp-connectivity?locale=en-US">SAP
+     * Connectivity Service (Cloud Foundry)</a>. This is the proxy service that enables applications to access
+     * On-Premise systems from the SAP Business Technology Platform (Cloud Foundry).
+     */
+    @Nonnull
+    public static final ServiceIdentifier CONNECTIVITY = of("connectivity");
+
+    /**
+     * Represents the <a href="https://api.sap.com/api/CFAuditLogRetrievalAPI/overview">SAP Audit Log Retrieval (Cloud
+     * Foundry environment)</a> service.
+     */
+    @Nonnull
+    public static final ServiceIdentifier AUDIT_LOG = of("auditlog-management");
```

Comment:
(Major)
How confident do you/we feel about maintaining the predefined set of identifiers?

Reply:
That's a good question.

I would like to note, though, that the provided string constants here are not necessarily connected to (i.e. the same as) anything that might be stored in the `ServiceBinding` (or the VCAP services structure).
It would be, for example, perfectly fine to define `ServiceIdentifier DESTINATION = of("foobar")` as long as we (or customers) would (somehow) be able to derive the correct `ServiceIdentifier` from a given `ServiceBinding`.
These constants would still remain valid and could be used for comparison, no matter how they are derived from a Service Binding - as long as it is done deterministically.